gr-blocks/python/blocks/sigmf_sink_minimal.py

```diff
@@ -0,0 +1,83 @@
+#
+# This file is part of GNU Radio
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+#
+
+from gnuradio import gr
+from . import blocks_python as blocks
+import numpy as np
+import json
+
+class sigmf_sink_minimal(gr.hier_block2):
+    """
+    A partial implementation of the SigMF standard to allow saving SigMF files from GNU Radio.
+    It is expected that this block will be replaced with gr-sigmf in the near future, and thus
+    it is immediately being depreciated.
+    For more information on SigMF see https://github.com/gnuradio/SigMF
+    
+    Args:
+        item_size: complex, float, or short is all that is supported at the moment
+        filename: filename, NOT including the extension, SigMF uses .sigmf-meta and .sigmf-data which will be automatically added to your filename
+        sample_rate: sample rate in Hz
+        center_freq: optional center Frequency in Hz
+        author: optional string used to record author of recording
+        description: optional string used to store whatever other info you want about the recording
+        hw_info: optional string used to record hardware used in making of recording, e.g. SDR type and antenna, or whether it's simulated data
+    """
+
+    def __init__(self, item_size, filename, sample_rate, center_freq, author, description, hw_info, is_complex):
+    
+        # Input type, which is included in .sigmf-meta file
+        if item_size == 8:
+            datatype_str = 'cf32_le'
+        elif item_size == 4:
+            datatype_str = 'rf32_le'
+        elif item_size == 2 and is_complex:
+            datatype_str = 'ci16_le'
+        elif item_size == 2 and not is_complex:
+            datatype_str = 'ri16_le'
+        else:
+            raise ValueError
+        
+        if '.sigmf' in filename:
+            filename = filename.rsplit('.', 1)[0]
+        gr.log.info('Generating '+filename+'.sigmf-meta, writing SigMF data to '+filename+'.sigmf-data')
+
+        gr.hier_block2.__init__(self, "sigmf_sink_minimal",
+                                gr.io_signature(1, 1, item_size),
+                                gr.io_signature(0, 0, 0))
+
+        file_sink = blocks.file_sink(item_size, filename + '.sigmf-data', False) # use regular File Sink, no append
+        self.connect(self, file_sink)
+        
+        # JSON/Meta Stuff
+        sigmf_version = "1.0.0" # update me if the time comes
+        
+        # Create .sigmf-meta file
+        meta_dict = {
+            "global": {
+                "core:datatype": datatype_str,
+                "core:sample_rate": float(sample_rate),
+                "core:hw": str(hw_info),
+                "core:author": str(author),
+                "core:description": str(description),
+                "core:version": sigmf_version
+            },
+            "captures": [
+                {
+                    "core:sample_start": 0,
+                }
+            ],
+            "annotations": []
+        }
+        if center_freq:
```

Comment:
hm, what if I explicitly want to set `center_freq` to 0, e.g. in a system where there's more or less well-centered quadrature mixing/close-to-zero-IF, I can see how that'd happen.

Recommendation: give the `center_freq` parameter a `= None` default, check for `is not None`.

Reply:
Couldn't figure out how to allow None for float type

Reply:
I know it was my suggestion to make this optional, since it is per the spec, but IMO we should just require this be populated if we can't also allow zero.

Reply:
BTW, the way you can do this is sort of complicated:

```diff
--- a/gr-blocks/grc/blocks_sigmf_sink_minimal.block.yml
+++ b/gr-blocks/grc/blocks_sigmf_sink_minimal.block.yml
@@ -21,7 +21,8 @@ parameters:
 -   id: center_freq
     label: Center Frequency
     dtype: float
-    default: 0
+    options: [np.nan]
+    option_labels: ['None']
 -   id: author
     label: Author
     dtype: string
@@ -43,7 +44,9 @@ asserts:
 - ${ len(filename) > 0 }
 templates:
-    imports: from gnuradio import blocks
+    imports: |-
+        from gnuradio import blocks
+        import numpy as np
     make: |-
         blocks.sigmf_sink_minimal(
             item_size=${type.size},
```

Then just check to see if the value is `np.nan` instead of `if frequency:`